File: 746_min_cost_climbing_stairs.py
class Solution:
    def minCostClimbingStairs(self, cost: List[int]) -> int:
        dpOne = cost[0]
        dpTwo = cost[1]
        
        for i in range(2, len(cost)):
            temp = cost[i] + min(dpOne, dpTwo)
            dpOne = dpTwo
            dpTwo = temp
        
        return min(dpOne, dpTwo)

File: swift/746_min_cost_climbing_stairs.swift
class Solution {
  func minCostClimbingStairs(_ cost: [Int]) -> Int {
    var minCosts: [Int] = Array(repeating: 0, count: cost.count)

    for index in 0..<cost.count {
      if index < 2 {
        minCosts[index] = cost[index]
      } else {
        minCosts[index] = min(cost[index] + minCosts[index - 1], cost[index] + minCosts[index - 2])
      }
    }

    return min(minCosts[cost.count - 2], minCosts[cost.count - 1])
  }
}
